mlir/test/Dialect/GEN/invalid.mlir

```diff
@@ -0,0 +1,17 @@
+// RUN: mlir-opt -split-input-file %s -verify-diagnostics
+
+func.func @test_3d_nd_range_bounds_low() {
+  %c-1 = arith.constant -1 : i32
+  // expected-error @below {{'gen.local_id' op input dimension must be in the range [0, 3). Got -1}}
+  %0 = gen.local_id %c-1
+  func.return
+}
+
+// -----
+
+func.func @test_3d_nd_range_bounds_high() {
+  %c3 = arith.constant 3 : i32
+  // expected-error @below {{'gen.work_group_id' op input dimension must be in the range [0, 3). Got 3}}
+  %0 = gen.work_group_id %c3
+  func.return
+}
```

Comment:
Okey, I think issue's I was doing this as "runtime checks". I changed it so that's checked at compile time and show that is and should not be possible. This "static verification" is done also in other interfaces ([see](https://github.com/intel/llvm/blob/b48e342a192060e66661292fafe798f161579a48/mlir/include/mlir/Interfaces/ControlFlowInterfaces.td#L264)).